### PR TITLE
Add discovery api

### PR DESCRIPTION
**Pull request by kcarra**
Squashed commits for all changes we made to integrate discovery_api into the umbrella excluding the files that were copied https://github.com/smartcitiesdata/smartcitiesdata/pull/470/commits/7bc2c3798b48177b7299ac7694472aae74569a0b

### CODE REVIEW

File: apps/odo/mix.exs [hunk 1]
```diff
@@ -30,7 +30,7 @@ defmodule Odo.MixProject do
       {:credo, "~> 1.0", only: :dev, runtime: false},
       {:distillery, "~> 2.1"},
       {:ex_aws, "~> 2.1"},
-      {:ex_aws_s3, "~> 2.0"},
+      {:ex_aws_s3, "~> 2.0", [env: :prod, git: "https://github.com/ex-aws/ex_aws_s3", ref: "6b9fdac73b62dee14bffb939965742f2576f2a7b"]},
```
Comment (mcsearchin) [inline apps/odo/mix.exs]: What happened here?  Do we need this very specific version that isn't available from hex?

File: apps/discovery_api/config/config.exs [hunk 2]
```diff
@@ -37,7 +37,7 @@ config :discovery_api, DiscoveryApi.Quantum.Scheduler,
 
 config :mime, :types, %{
   "application/zip" => ["zip", "shp", "shapefile"],
-  "application/gtfs" => ["pb"]
+  "application/gtfs+protobuf" => ["gtfs"]
```
Comment (mcsearchin) [inline apps/discovery_api/config/config.exs]: Do we know how this will impact Discovery UI?

File: apps/odo/mix.exs (hunk 1 above)
Comment (kcarra) [reply apps/odo/mix.exs]: Yeah, it is needed by discovery_api in order to use streaming. Still pending release. I'll add a comment to make that clear.

File: apps/discovery_api/config/config.exs (hunk 2 above)
Comment (bbrewer-papa-x) [reply apps/discovery_api/config/config.exs]: Yeah, I'm interested to know. Probably not your fault if it breaks stuff b/c our mime type stuff is a dumpster show

File: apps/discovery_api/mix.exs [hunk 3]
```diff
@@ -58,7 +62,7 @@ defmodule DiscoveryApi.Mixfile do
       {:phoenix_pubsub, "~> 1.0"},
       {:nanoid, "~> 2.0"},
       {:placebo, "~> 1.2.2", only: [:dev, :test]},
-      {:plug_cowboy, "~> 1.0"},
+      {:plug_cowboy, "~> 2.1"},
```
Comment (bbrewer-papa-x) [inline apps/discovery_api/mix.exs]: Have we confirmed if the hosted file download bug (memory overflow) is present with this? I don't think we have enough of them for it to matter if not.

File: apps/odo/mix.exs (hunk 1 above)
Comment (bbrewer-papa-x) [reply apps/odo/mix.exs]: Yeah, also, this is the one that overflows cowboy when you try to stream hosted files, btw

File: apps/discovery_api/config/config.exs (hunk 2 above)
Comment (kcarra) [reply apps/discovery_api/config/config.exs]: the issue is the TDG doesn't accept `application/gtfs` anymore so if we keep it how it is then the TDG returns `application/octet-stream` or whatever the default sourceFormat is. I did grep for any datasets in prod, staging and dev with the `application/gtfs` source format and it doesn't look like there are any.

File: apps/discovery_api/test/unit/discovery_api/data/mapper_test.exs [hunk 4]
```diff
@@ -28,7 +28,7 @@ defmodule DiscoveryApi.Data.MapperTest do
         [%{business: %{conformsToUri: nil}}, [:conformsToUri], "https://project-open-data.cio.gov/v1.1/schema/"],
         [%{business: %{conformsToUri: "overridden"}}, [:conformsToUri], "https://project-open-data.cio.gov/v1.1/schema/"],
         [%{technical: %{sourceFormat: "application/json"}}, [:fileTypes], ["JSON"]],
-        [%{technical: %{sourceFormat: "application/gtfs"}}, [:fileTypes], ["JSON"]],
+        [%{technical: %{sourceFormat: "application/gtfs+protobuf"}}, [:fileTypes], ["GTFS"]],
```
Comment (LtChae) [inline apps/discovery_api/test/unit/discovery_api/data/mapper_test.exs]: I would expect fileTypes here to have "JSON", not GTFS. I believe this was covered with an override somewhere in the bit that generates the file types.

**Pull request merged**